manifests/init.pp

```diff
@@ -14,12 +14,16 @@
   $write_queue_limit_low  = undef,
   $package_name           = $collectd::params::package,
   $version                = installed,
+  $minimum_version        = undef,
 ) inherits collectd::params {
 
   $plugin_conf_dir = $collectd::params::plugin_conf_dir
   validate_bool($purge_config, $fqdnlookup)
   validate_array($include, $typesdb)
 
+  # Version for templates
+  $_collectd_version = pick($::collectd_version, regsubst($version, '^\d+(?:\.\d+){1.2}', '\0'), $minimum_version)
```

Comment:
The regex is only meant to match a version number of collectd (as much of the major, minor and patch parts that are supplied). it returns undef if version happens to be 'installed', 'present' or 'absent'. validate_re is not appropriate in this circumstance.

re: fact name
I don't mind changing the fact name if that's what is preferred. I changed the templates because I was concerned that the fact was considered a part of the modules 'public' interface.
Make a decision either way and let me know.